src/computational_graph/graph.jl

```diff
@@ -112,11 +78,12 @@ mutable struct Graph{F,W} # Graph
     - `factor`  additional factor of the diagram
     - `weight`  weight of the diagram
     """
-    function Graph(vertices::Vector{OperatorProduct}; external=[], subgraph=[],
+    function Graph(vertices::AbstractVector; external=collect(1:length(vertices)), subgraph=[], topology=[],
```

Comment:
Julia provides a builtin function `eachindex` for this pattern, so we should refactor `1:length(A)` to `eachindex(A)` throughout.

Reply:
fixed